### PR TITLE
Cw 237 enhance error catching reporting

### PR DESCRIPTION
**Pull request by OmarHatem28**
- Save Errors/Exceptions into a local file
- Allow the user to select either send the error logs or ignore
Screen Shot 2023-01-09 at 8 05 56 PM

### CODE REVIEW

File: lib/src/screens/failure_page.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+              ),
+              Padding(
+                padding: const EdgeInsets.symmetric(vertical: 20),
+                child: Text(
+                  "Oops, we got some error.",
+                  style: theme.textTheme.headline1?.copyWith(fontSize: 20),
+                ),
+              ),
+              Text(
+                "Please send crash report to our support team to make the application better.",
```
Comment (SamsungGalaxyPlayer) [inline lib/src/screens/failure_page.dart]: Please send the crash report to our support team to make the application better.

(add `the`)

File: lib/main.dart [hunk 2]
```diff
@@ -210,6 +188,34 @@ void _saveException(String? error, StackTrace? stackTrace) async {
...
+  print(details.exception.toString());
+  WidgetsBinding.instance.addPostFrameCallback(
+      (timeStamp) {
+        showPopUp<void>(
+          context: navigatorKey.currentContext!,
+          builder: (context) {
+            return AlertWithTwoActions(
+              isDividerExist: true,
+              alertTitle: S.of(context).error,
+              alertContent: "Oops, we got some error.\n\nPlease send crash report to our support team to make the application better.",
```
Comment (SamsungGalaxyPlayer) [inline lib/main.dart]: Please send the crash report to our support team to make the application better.

(add `the`)

File: res/values/strings_en.arb [hunk 3]
```diff
@@ -679,5 +679,7 @@
   "tor_only": "Tor only",
   "unmatched_currencies": "Your current wallet's currency does not match that of the scanned QR",
   "contact_list_contacts": "Contacts",
-  "contact_list_wallets": "My Wallets"
+  "contact_list_wallets": "My Wallets",
+  "do_not_send": "Don't send",
+  "error_dialog_content": "Oops, we got some error.\n\nPlease send crash report to our support team to make the application better."
```
Comment (SamsungGalaxyPlayer) [inline res/values/strings_en.arb:684]: Please send the crash report to our support team to make the application better.

(add `the`)

**Review by SamsungGalaxyPlayer — CHANGES_REQUESTED** [review]

File: lib/src/screens/failure_page.dart (hunk 1 above)
Comment (OmarHatem28) [reply lib/src/screens/failure_page.dart]: this page was removed!! can you please make sure you are reviewing the latest commit

Comment (OmarHatem28) [reply lib/src/screens/failure_page.dart]: but yeah the wording it self would be fixed

**Review by mkyq — APPROVED** [review]

**Review by SamsungGalaxyPlayer — APPROVED** [review]

**Pull request merged**